test/regression/convert/covert_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+## registry+v1 bundle regression test
+
+This test verifies that rendering registry+v1 bundles to manifests
+always produces the same files and content.
+
+It runs:     go run generate-manifests.go -output-dir=./testdata/tmp/rendered/
+Then compares: ./testdata/tmp/rendered/ vs ./testdata/expected-manifests/
+
+Files are sorted by kind/namespace/name for consistency.
+
+To update expected output (only on purpose), run:
+
+	go run generate-manifests.go -output-dir=./testdata/expected-manifests/
+*/
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Test_RenderedOutputMatchesExpected runs generate-manifests.go,
+// then compares the generated files against expected-manifests/.
+//
+// It fails if any file differs or is missing.
+//
+// TMP dir is cleaned up after test ends.
+func Test_RenderedOutput_MatchesExpected(t *testing.T) {
+	tmpRoot := "./testdata/tmp/rendered/"
```

Comment:
Hard coding `tmpRoot := "./testdata/tmp/rendered"` writes into your source tree. Consider using `t.TempDir()` (or at least `os.MkdirTemp`) so failures don’t leave junk under testdata/ and so tests can run in parallel without colliding.

Reply:
Done :thanks: